src/inet/UDPEndPointImplLwIP.cpp

```diff
@@ -133,7 +133,7 @@ CHIP_ERROR UDPEndPointImplLwIP::LwIPBindInterface(struct udp_pcb * aUDP, Interfa
 InterfaceId UDPEndPointImplLwIP::GetBoundInterface() const
 {
 #if HAVE_LWIP_UDP_BIND_NETIF
-    return InterfaceId(netif_get_by_index(mUDP->netif_idx));
+    return InterfaceId(netif_get_by_index(mUDP->netif_idx + 1));
```

Comment:
This does not seem like the right fix.  Either we should have the right interface index here, or if this is supposed to be "wildcard" then we should be doing something to expand the wildcard.  But just blindly adding 1 does not make sense.

Reply:
https://github.com/project-chip/connectedhomeip/blob/9d2eeb052310eee86b19bec413790fe61712ebb7/third_party/lwip/repo/lwip/src/include/lwip/netif.h#L507
Following this logic, I arrived at this conclusion.

Reply:
Yes, but we're supposed to be storing an interface index _already_, no?  So we should not be storing 0.  Why are we storing 0?

Reply:
I am not an lwip expert but it seems `netif_idx` only gets set once a bind call is done. Is it possible that `LwIPBindInterface` sets the `mUDP->netif_idx` to 0? There might be underlying issue that does not set the `netif_idx` correctly.

As temporary fix could be to use `num` attribute and create a seperate issue to verify the lwip implementation.
```
index = netif_get_index(mUDP)
netif_get_by_index(index)
```

Reply:
@bzbarsky-apple could leaving the one as a temp fix to enable groups but acceptable with a an issue to revisit the lwip implementation?

Reply:
The thing is.. this fix makes no sense.  Why do we think the right interface is +1 and not +2, or +0 if we already have a nonzero thing?

Reply:
That is, I have no indication that this even fixes more than it breaks.